### ISSUE TITLE
[Translation] 6_epoll_kqueue_iocp

### ISSUE THREAD

**Issue by Johnny4Fun** [cfsamson/book-exploring-async-basics#31]
> `Mio` powers the OS event queue used in [Tokio](https://github.com/tokio-rs/tokio), **which** is a runtime that provides I/O, networking, scheduling etc.

Here the word `which` refers to `Tokio`, right? So does it matter whether there is a comma before `which`? To my knowledge, the answer is NO, what a `which` refers to just depends on the context.

> This is especially apparent when you want to write a library since abstracting over both means you'll either have to model `IOCP` as `readiness-based` or model `epoll/kqueue` as completion-based.

Can I understand the word `model` here as `abstract` or `transform`?

**Comment by cfsamson** [comment]
> Here the word which refers to Tokio, right? So does it matter whether there is a comma before which? To my knowledge, the answer is NO, what a which refers to just depends on the context.

Yes, it refers to Tokio. I think the comma is correct as is since it follows a fully qualified sentence, but I have to add that English comma rules is something I get wrong sometimes.

>Can I understand the word model here as abstract or transform?

I think abstract could be used here. Transform is not strictly correct since all you provide is an abstraction that covers up the fact that they're different.
